### PR TITLE
Bugfix/lattice 2359 search maxhits

### PR DESCRIPTION
**Pull request by you-leee**

### CODE REVIEW

File: src/test/java/com/openlattice/rehearsal/search/SearchControllerTest.kt [hunk 1]
```diff
@@ -373,12 +411,104 @@ class SearchControllerTest : MultipleAuthenticatedUsersBase() {
...
+        var searchedEntities = searchApi.executeEntitySetDataQuery(es.id, searchMax0)
+
+        // should yield 0 hits and 0 totalHits
+        Assert.assertEquals(0, searchedEntities.hits.size)
+        Assert.assertEquals(numberOfEntities, searchedEntities.numHits.toInt())
+
+        val searchMax1 = SearchTerm("*", 0, 1)
+        searchedEntities = searchApi.executeEntitySetDataQuery(es.id, searchMax1)
+
+        // should yield 1 hit, but 10 totalHits
```
Comment (dbt) [inline src/test/java/com/openlattice/rehearsal/search/SearchControllerTest.kt:442]: tiny issue but the 10 in this comment is incorrect. :)

File: src/test/java/com/openlattice/rehearsal/search/SearchControllerTest.kt [hunk 2]
```diff
@@ -373,12 +411,104 @@ class SearchControllerTest : MultipleAuthenticatedUsersBase() {
...
+            Assert.assertTrue(entities.keys.contains(id))
+        }
+
+        // Delete multiple entities
+        val deletedIds = (0 until 50).map {
+            idsList[Random.nextInt(numberOfEntities)]
+        }.toSet()
+        val numberOfDeletes = deletedIds.size // could pick same id
+
+        dataApi.deleteEntities(es.id, deletedIds.toSet(), DeleteType.values()[Random.nextInt(2)])
```
Comment (dbt) [inline src/test/java/com/openlattice/rehearsal/search/SearchControllerTest.kt]: deletedIds is already a set, so toSet is redundant.

re: DeleteType, I would say either pick one or do some with each delete type. If there's ever a bug with only soft types, for example, the test would start failing 50% of the time. So you could split up the IDs into two batches and delete some hard and some soft.

**Review by dbt — CHANGES_REQUESTED** [review]
yay for tests

File: src/test/java/com/openlattice/rehearsal/search/SearchControllerTest.kt (hunk 1 above)
Comment (you-leee) [reply src/test/java/com/openlattice/rehearsal/search/SearchControllerTest.kt:442]: Right.. thanks! I updated `numberOfEntities` to 20, but then I made a different test for deleting + maxHit limited search. Will change it back to 10

File: src/test/java/com/openlattice/rehearsal/search/SearchControllerTest.kt (hunk 2 above)
Comment (you-leee) [reply src/test/java/com/openlattice/rehearsal/search/SearchControllerTest.kt]: Good idea! The behaviour here should be the same for both delete types.

**Review by dbt — APPROVED** [review]

**Review by pckeyes — APPROVED** [review]

**Review by makosblade — APPROVED** [review]

**Pull request merged**